### PR TITLE
Add in an edit button to Collection page

### PR DESCRIPTION
**Pull request by lamtu1**
Add in a button to go to the `edit` section for Collection rather than going to the `Dashboard`.

### CODE REVIEW

**Comment by lamtu1** [comment]
fixes #1719

**Comment by lamtu1** [comment]
`SCREENSHOT:` Showing the `edit` button in Collection

Screenshot 2023-08-14 at 4 47 08 PM

File: app/views/hyrax/collections/show.html.erb [hunk 1]
```diff
@@ -90,7 +90,7 @@
   <%= render 'shared/colored_divider', color: 'lime-green' do %>
     <a href="<%= view_all_url %>" class='btn btn-lg btn-primary collection-landing-button col-sm-10'>View All</a>
     <%# NEW BUTTON: Add in an edit button to collection rather than going through the dashboard %>
-    <% if user_signed_in? && current_user.admin? %>
+    <% if user_signed_in? && (current_user.admin? || current_ability.can?(:edit, @presenter.solr_document)) %>
```
Comment (CGillen) [inline app/views/hyrax/collections/show.html.erb:93]: Patron users can also create collections & don't have access to the dashboard to edit their collections. You need to make sure this isn't a User Collection type collection. We usually do that with something like:
`collection_type.machine_id == 'user_collection'` You'll have to edit that to properly get the collection_type

**Review by CGillen — CHANGES_REQUESTED** [review]

File: app/views/hyrax/collections/show.html.erb (hunk 1 above)
Comment (lamtu1) [reply app/views/hyrax/collections/show.html.erb:93]: @CGillen How does the `user_collection` see on their end when they edit? Is it just this?

Screenshot 2023-08-18 at 10 30 21 AM

**Review by CGillen — APPROVED** [review]

**Pull request merged**